exec/java-exec/src/main/java/org/apache/drill/exec/work/filter/RuntimeFilterSink.java

```diff
@@ -74,32 +70,43 @@
 
   private SendingAccountor sendingAccountor;
 
+  private  AsyncAggregateWorker asyncAggregateWorker;
+
+  private AtomicBoolean running = new AtomicBoolean(true);
+
   private static final Logger logger = LoggerFactory.getLogger(RuntimeFilterSink.class);
 
 
   public RuntimeFilterSink(DrillbitContext drillbitContext, SendingAccountor sendingAccountor)
   {
     this.drillbitContext = drillbitContext;
     this.sendingAccountor = sendingAccountor;
-    AsyncAggregateWorker asyncAggregateWorker = new AsyncAggregateWorker();
-    drillbitContext.getExecutor().submit(asyncAggregateWorker);
   }
 
   public void add(RuntimeFilterWritable runtimeFilterWritable)
   {
+    if (asyncAggregateWorker == null) {
+      asyncAggregateWorker = new AsyncAggregateWorker();
+      drillbitContext.getExecutor().submit(asyncAggregateWorker);
+    }
```

Comment:
Should be done once in constructor. Putting here has issues such that multiple netty threads can create multiple `asyncAggregateWorker` thread and leak all except the last set one.